plugins/catalog-backend/src/ingestion/types.ts

```diff
@@ -65,3 +65,11 @@ export type ReadLocationError = {
   location: LocationSpec;
   error: Error;
 };
+
+//
+// ConfigGenerator
+//
+
+export type ConfigGenerator = {
+  generateConfig(repoPath: string): Promise<Entity[]>;
```

Comment:
yeah so probably renaming this according to the above comment, and taking something like an `AnalyzeLocationRequest` and returning `Promise<AnalyzeLocationResponse>`

Reply:
Changed